components/net/bluetooth_thread.rs

```diff
@@ -395,44 +401,66 @@ impl BluetoothManager {
         sender.send(message).unwrap();
     }
 
-    pub fn get_primary_service(&mut self, device_id: String, sender: IpcSender<BluetoothObjectMsg>) {
+    pub fn get_primary_service(&mut self, device_id: String, uuid: String, sender: IpcSender<BluetoothObjectMsg>) {
         let mut adapter = match self.get_adapter() {
             Some(a) => a,
             None => send_error!(sender, "No adapter found"),
         };
+        let primary_service = match self.get_gatt_service_by_uuid(&mut adapter, &device_id, &uuid) {
+            Some(s) => s,
+            None => send_error!(sender, "No primary service found")
+        };
+        if primary_service.is_primary().unwrap_or(false) {
+            let message = BluetoothObjectMsg::BluetoothService {
+                uuid: primary_service.get_uuid().unwrap_or("".to_owned()),
+                is_primary: true,
+                instance_id: primary_service.get_object_path(),
+            };
+            sender.send(message).unwrap();
+            return;
```

Comment:
maybe we should check if !primary {send_error!}
and than we won't need the `return;` line